### PR TITLE
Issue 679  collect ignore

### PR DESCRIPTION
**Pull request by kohr-h**
Fix for #679, works in my working environment, but I'll test it with more configurations. Windows tests are also appreciated :-)

### CODE REVIEW

**Comment by kohr-h** [comment]
Works also on fresh environments for me.

**Comment by adler-j** [comment]
Does `pytest` work?

**Comment by kohr-h** [comment]
Yes

**Comment by adler-j** [comment]
No need for `pytest odl`?

**Comment by kohr-h** [comment]
No, `pytest.ini` does the job.

File: .travis.yml [hunk 1]
```diff
@@ -51,6 +51,8 @@ before_script:
 script:
   # Run tests, including PEP8 check, and produce a coverage report
   - py.test --doctest-modules --cov --cov-report term-missing --pep8 $TRAVIS_BUILD_DIR/odl
+  # Invoke also the alternative way of running the unit tests
```
Comment (adler-j) [inline .travis.yml]: Does this cause travis to run twice? In that case i'm sceptical, that will take some time.

Comment (kohr-h) [inline .travis.yml]: We're talking 15 extra seconds here.

Comment (adler-j) [inline .travis.yml]: I guess we can live with that

Comment (kohr-h) [inline .travis.yml]: I mean, it doesn't have to install everything again. It's just running the tests again, which is somewhat negligible to the 2 1/2 minutes (?) of the overall procedure.

Comment (adler-j) [inline .travis.yml]: Sounds OK to me, at least for now.

**Comment by adler-j** [comment]
Any chance we can have a look at #688 and #677 while looking at this?

#677 can be solved by simply adding a line to the conftest that checks the pytest version (needs to be >=3.0.3) and throws an error otherwise.

# 688 may actually be fixed by this, investigating.

**Comment by adler-j** [comment]
So it seems this does not solve the problem at all for me, [I get this log when running the tests](https://gist.github.com/adler-j/176a5aef9dad0a761e5c455351edcaee).

It seems it cannot find the conftest file.

I get the same error regardless of how I attempt to run the code:

* `pytest`
* `pytest odl`
* `python -c "import odl; odl.test()"`

**Comment by adler-j** [comment]
Looking at this some more, I'm failing to understand by what mechanism pytest is supposed to find the `conftest` with this setup in this branch. How does it work?

**Comment by adler-j** [comment]
Success!

It seemed I needed to run `pip install -e .` in order for the pytest hook to be installed. It now works on my machine. Looking at #677 and #688 now

**Comment by adler-j** [comment]
#688 solved in a simple manner, will see if #677 is as simple

**Comment by adler-j** [comment]
#677 is not as straightforward. One option is to add:

```python
def pytest_configure(config):
    import pytest
    print(dir(config))
    if pytest.__version__ < '3.0.3':
        raise ImportError('ODL tests require pytest version >=3.0.3')
```

to `pytest_plugins.py`, but this gives ugly errors:

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "d:\anaconda\lib\site-packages\_pytest\main.py", line 92,in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "d:\anaconda\lib\site-packages\_pytest\config.py", line 859, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "d:\anaconda\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 750, in call_historic
INTERNALERROR>     self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "d:\anaconda\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "d:\anaconda\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "d:\anaconda\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 614, in execute
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "d:\github\odl\odl\util\pytest_plugins.py", line 46, in pytest_configure
INTERNALERROR>     raise ImportError('ODL tests require pytest version >=3.0.3')
INTERNALERROR> ImportError: ODL tests require pytest version >=3.0.3
```

Is this acceptable anyway? The error at the bottom is actually quite descriptive, only the `INTERNALERROR` is ugly.

**Comment by adler-j** [comment]
After that is resolved I guess we can go ahead.

**Comment by kohr-h** [comment]
Sorry for being quiet, busy day. Anyway, good that things work out consistently now. I would be okay with the above error mechanism, there doesn't seem to be a better way of doing it. I guess raising an error at any place in a pytest hook will invoke this INTERNALERROR stack.

**Comment by kohr-h** [comment]
Can we use [this](http://doc.pytest.org/en/latest/customize.html#confval-minversion) in the `pytest.ini`?

**Comment by kohr-h** [comment]
With the `pytest.ini` entry, we get the following error:
```
ERROR: /home/hkohr/git/odl/pytest.ini:2: requires pytest-3.0, actual pytest-2.9.2'
```
Looks a bit nicer, eh?

**Comment by adler-j** [comment]
3.0 is insufficient, we need 3.0.3, except for that that is exactly what I was looking for! After fixing that we can merge imo, but we gotta fix that pesky test that fails.

**Comment by kohr-h** [comment]
> 3.0 is insufficient, we need 3.0.3

Okay, what's the particular reason? Just curious.
I'll put in 3.0.3 in the meanwhile and rebase. I'll also look at the test.

**Comment by kohr-h** [comment]
The failing test was a failure of `apt-get update`, nothing to do there.

**Comment by kohr-h** [comment]
Okay, got the last changes in, waiting for the test to succeed, but after that I guess I'll merge.

**Comment by adler-j** [comment]
We should make a 0.5.2 release for this btw.

**Comment by kohr-h** [comment]
I'll do it in connection with the merge of #696.

**Pull request merged**